packages/utils-react/src/use-reapit-data/use-reapit-update.ts

```diff
@@ -76,18 +79,22 @@ export const send =
 
     if (!getHeaders) throw new Error('Missing valid Reapit Connect Session')
 
+    console.log('starting')
+
     await setLoading(true)
     setSuccess(undefined)
     try {
+      console.log('fetching')
+
       const response = await fetch(url, {
         headers: getHeaders,
         method,
         body: JSON.stringify(params),
       })
 
-      if (!returnUpdatedModel) await setLoading(false)
+      if (!returnUpdatedModel && !returnUpdatedModelWithLocation) await setLoading(false)
```

Comment:
There really is a lot of logic in here - is there a way to break it out into something more maintainable / testable? I am just thinking it's something we will use all over the place and feels a bit fragile / complex